### PR TITLE
ADEN-3947 Add missing slots to IndexExchange configuration

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/lookup/prebid/adapters/indexExchange.js [hunk 1]
```diff
@@ -40,6 +120,20 @@ define('ext.wikia.adEngine.lookup.prebid.adapters.indexExchange',[
 		return geo.isProperGeo(instantGlobals.wgAdDriverIndexExchangeBidderCountries);
 	}
 
+	function getSlots(skin, pageType) {
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/lookup/prebid/adapters/indexExchange.js]: how about doing something similar like in openxbidder?

``` javascript
    function setupHomePageSlots() {
        var slots = getSlots();

        Object.keys(slots).forEach(function (slotName) {
            if (slotName.indexOf('TOP') > -1) {
                addSlot('HOME_' + slotName, slots[slotName]);
                removeSlot(slotName);
            }
        });

        addSlot('PREFOOTER_MIDDLE_BOXAD', {sizes: ['300x250']});
    }
```

maybe this logic can be extracted an shared among bidders? We would need to add `addSlot` and `removeSlot`

**Review by bognix — COMMENTED** [review]
How about adding test case to indexExchange.spec.js to check the getSlots method in case of different pageType?

File: extensions/wikia/AdEngine/js/run/desktop.run.js [hunk 2]
```diff
@@ -102,6 +103,7 @@ require([
 	require.optional('ext.wikia.adEngine.slot.revcontentSlots')
 ], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/run/desktop.run.js:104]: MAJOR Function has 11 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/ARecoveryEngine/js/helper.js [hunk 3]
```diff
@@ -67,7 +67,7 @@ define('ext.wikia.aRecoveryEngine.recovery.helper', [
 					xmlHttp.send();
 				} catch (e) {}
```
Comment (WikiaTech) [inline extensions/wikia/ARecoveryEngine/js/helper.js:68]: MAJOR Either remove or fill this block of code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEmptyBlock)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 10 issues
- MAJOR 10 major

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [adContext.js#L14](https://github.com/Wikia/app/blob/fe52579f1340b80f81b211d6e6b8f7b4499091df/extensions/wikia/AdEngine/js/context/adContext.js#L14): Function has 8 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
2. MAJOR [adContext.js#L43](https://github.com/Wikia/app/blob/fe52579f1340b80f81b211d6e6b8f7b4499091df/extensions/wikia/AdEngine/js/context/adContext.js#L43): Function has a complexity of 48 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
3. MAJOR [adContext.js#L139](https://github.com/Wikia/app/blob/fe52579f1340b80f81b211d6e6b8f7b4499091df/extensions/wikia/AdEngine/js/context/adContext.js#L139): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1067)
4. MAJOR [desktop.run.js#L22](https://github.com/Wikia/app/blob/fe52579f1340b80f81b211d6e6b8f7b4499091df/extensions/wikia/AdEngine/js/run/desktop.run.js#L22): Function has 18 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
5. MAJOR [AdContext.spec.js#L26](https://github.com/Wikia/app/blob/fe52579f1340b80f81b211d6e6b8f7b4499091df/extensions/wikia/AdEngine/js/spec/AdContext.spec.js#L26): Function has a complexity of 11 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
6. MAJOR [openXBidder.spec.js#L7](https://github.com/Wikia/app/blob/fe52579f1340b80f81b211d6e6b8f7b4499091df/extensions/wikia/AdEngine/js/spec/lookup/openXBidder.spec.js#L7): Remove this useless assignment to local variable "slotParams" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)
7. MAJOR [openXBidder.spec.js#L139](https://github.com/Wikia/app/blob/fe52579f1340b80f81b211d6e6b8f7b4499091df/extensions/wikia/AdEngine/js/spec/lookup/openXBidder.spec.js#L139): Remove this useless assignment to local variable "slotParams" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)
8. MAJOR [Ooyala.js#L78](https://github.com/Wikia/app/blob/fe52579f1340b80f81b211d6e6b8f7b4499091df/extensions/wikia/VideoHandlers/js/handlers/Ooyala.js#L78): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ACommentedCode)

**Pull request closed without merging**